driver/sql/projection_notification_processor.go

```diff
@@ -101,31 +164,38 @@ func (b *projectionNotificationProcessor) Start(ctx context.Context, handler Pro
 
 // Queue puts the notification on the queue to be processed
 func (b *projectionNotificationProcessor) Queue(ctx context.Context, notification *ProjectionNotification) error {
-	select {
-	default:
-	case <-ctx.Done():
-		return context.Canceled
-	case <-b.done:
-		return errors.New("goengine: unable to queue notification because the processor was stopped")
-	}
-
-	b.metrics.QueueNotification(notification)
+	return b.notificationQueue.Queue(ctx, notification)
+}
 
-	b.queue <- notification
-	return nil
+// ReQueue puts the notification again on the queue to be processed with a ValidAfter set
+func (b *projectionNotificationProcessor) ReQueue(ctx context.Context, notification *ProjectionNotification) error {
+	return b.notificationQueue.ReQueue(ctx, notification)
 }
 
 func (b *projectionNotificationProcessor) startProcessor(ctx context.Context, handler ProcessHandler) {
+ProcessorLoop:
 	for {
 		select {
 		case <-b.done:
 			return
 		case <-ctx.Done():
 			return
 		case notification := <-b.queue:
```

Comment:
Question but would move [`b.queue`](https://github.com/hellofresh/goengine/pull/41/files#diff-6a6ba4cfc9a5e4a9e7d0aac85e41f166R21) into `notificationQueue` not be a little nicer?